### ISSUE TITLE
Bedrock Provider using profile

### ISSUE THREAD

**Issue by liorheber** [vercel/ai#2053]
### Feature Description

When working in a dev environment, we would often use AWS session tokens for expiring access under a profile.
This is supported in AWS clients as seen here - https://docs.aws.amazon.com/cli/v1/userguide/cli-configure-files.html

It would be great if the provider would maintain this capability.

### Use Case

Working with expiring access and multiple profiles during development environment.

```
const bedrock = createAmazonBedrock({
  region: "us-east-1",
  profile: "LAB-OKTA",
});
```

**Comment by lgrammel** [comment]
Good point! I'll think about how to expose the full range of authentication options, while keeping the most basic one as easy as possible

**Comment by skorfmann** [comment]
get this error in a AWS Lambda context with implicit role credentials from the Lambda function. So that's essentially the session token missing here as well, I believe.

```{
  "errorType": "Error",
  "errorMessage": "The security token included in the request is invalid.",
  "trace": [
    "Error: The security token included in the request is invalid.",
    "    at Runtime.handler (/var/task/index.js:6720:11)",
    "    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)"
  ]
}
```

**Comment by lgrammel** [comment]
https://github.com/vercel/ai/pull/2074

**Comment by lgrammel** [comment]
Note sure if this will fully address this issue, since I did not find `profile` in `BedrockRuntimeClientConfig`, but this is the level of integration that I want to support: #2074 - you can pass any custom client config.

**Comment by skorfmann** [comment]
> Note sure if this will fully address this issue, since I did not find `profile` in `BedrockRuntimeClientConfig`, but this is the level of integration that I want to support: #2074 - you can pass any custom client config.

I think this will do. This allows passing in https://docs.aws.amazon.com/AWSJavaScriptSDK/v3/latest/Package/-aws-sdk-credential-providers/#fromini to the client

**Comment by liorheber** [comment]
Thank you for the quick solution!

**Comment by airtonix** [comment]
> > Note sure if this will fully address this issue, since I did not find `profile` in `BedrockRuntimeClientConfig`, but this is the level of integration that I want to support: #2074 - you can pass any custom client config.
> 
> I think this will do. This allows passing in [docs.aws.amazon.com/AWSJavaScriptSDK/v3/latest/Package/-aws-sdk-credential-providers/#fromini](https://docs.aws.amazon.com/AWSJavaScriptSDK/v3/latest/Package/-aws-sdk-credential-providers/#fromini) to the client

I'm still getting an issue with regards to using aws sso profiles to access bedrock.

```jsonc
...
    "@ai-sdk/amazon-bedrock": "^0.0.3",
    "@ai-sdk/openai": "^0.0.34",
    "@ai-sdk/provider": "0.0.11",
    "ai": "^3.2.14",
    "ollama-ai-provider": "^0.7.0",
...
```

![image](https://github.com/vercel/ai/assets/61225/d6c2a3f0-bacf-4038-b805-c407f6a7bbdf)

I debug everything, and the secret key, etc all available as envvars, but i still get

```UnrecognizedClientException: The security token included in the request is invalid.```

**Comment by airtonix** [comment]
ok maybe I misinterpreted what is required. 

@lgrammel I see you've marked it as complete. (as a result of https://github.com/vercel/ai/pull/2074)

does this mean if we need to use a session token that we then are expected to own all the pieces of relicating this in our own code: 

https://github.com/vercel/ai/blob/77df9f0bbbba93632b486b2072d5cbd3cbe2c18c/packages/amazon-bedrock/src/bedrock-provider.ts#L49-L69

**Comment by airtonix** [comment]
ok yep... I had to provide this: 

```ts

    this.bedrock = createAmazonBedrock({
      bedrockOptions: {
        region: loadSetting({
          settingValue: this.options.region,
          settingName: 'region',
          environmentVariableName: 'AWS_REGION',
          description: 'AWS region',
        }),
        credentials: {
          accessKeyId: loadSetting({
            settingValue: this.options.accessKeyId,
            settingName: 'accessKeyId',
            environmentVariableName: 'AWS_ACCESS_KEY_ID',
            description: 'AWS access key ID',
          }),
          secretAccessKey: loadSetting({
            settingValue: this.options.secretAccessKey,
            settingName: 'secretAccessKey',
            environmentVariableName: 'AWS_SECRET_ACCESS_KEY',
            description: 'AWS secret access key',
          }),
          sessionToken: loadSetting({
            settingValue: this.options.sessionToken,
            settingName: 'sessionToken',
            environmentVariableName: 'AWS_SESSION_TOKEN',
            description: 'AWS secret access key',
          }),
        },
      },
    });
```

**Comment by lgrammel** [comment]
@airtonix yes you need to provide the `bedrockOptions` that you otherwise would use directly w/ Bedrock. However, your version has `this.options` which is not valid in this context. You could try the following (which does not have error handling though):

```ts
const bedrock = createAmazonBedrock({
  bedrockOptions: {
    region: process.env.AWS_REGION!,
    credentials: {
      accessKeyId: process.env.AWS_ACCESS_KEY!,
      secretAccessKey: process.env.AWS_SECRET_ACCESS_KEY,
      sessionToken: process.env.AWS_SESSION_TOKEN!
    },
  },
});
```

**Comment by airtonix** [comment]
Oh the soul less abandonment to `!`.

🥺🥹😭

All good. I just replicated what you did internally.

**Comment by lgrammel** [comment]
## Identify issue type

Identified as **Bug**  
Confidence: **High**

### Rationale
Although framed as an authentication enhancement, the comments document concrete failures with standard temporary AWS credentials because the session token was not handled by the existing provider configuration. Under the classification rules, the demonstrated broken behavior takes precedence over the feature framing.

### Evidence
- The original request asks to support AWS profiles and expiring session credentials in the Bedrock provider.
- A commenter reports a concrete Lambda failure with implicit role credentials: “The security token included in the request is invalid.”
- Another commenter reproduces the invalid-token error with AWS SSO credentials and resolves it by explicitly passing the session token through `bedrockOptions`.
